### PR TITLE
E2E & Node E2E: Move pods test into common directory

### PR DESCRIPTION
**Pull request by Random-Liu**
This is the 4th part of #29494.

For #29081.
Based on #29092, #29806.

The first commit is squash of all dependent commits. Please only review the last 2 commits.

The 2nd commit migrates pods.go to `common/pods.go`.
Notice that the test `should be schedule with cpu and memory limits` is removed because:
- It doesn't make sense at the node level.
- It should have been tested in scheduler_predicates at the cluster level https://github.com/kubernetes/kubernetes/blob/master/test/e2e/scheduler_predicates.go#L264

The 3rd commit splits pods.go into several pods (nothing is changed, only move code):
- **Liveness probe test:** Moved into `container_probe.go`.
- **Init container test:** Moved into a new file `init_container.go`.
- **Others:** Still in pods.go

@vishh @timstclair

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/29814)

### CODE REVIEW

**Comment by Random-Liu** [comment]
@vishh @timstclair PR ready for review! :)

File: test/e2e/common/container_probe.go [hunk 1]
```diff
@@ -90,6 +90,145 @@ var _ = framework.KubeDescribe("Probing container", func() {
...
+									Command: []string{"cat", "/tmp/health"},
+								},
+							},
+							InitialDelaySeconds: 15,
+							FailureThreshold:    1,
+						},
+					},
+				},
+			},
+		}, 1, defaultObservationTimeout)
```
Comment (timstclair) [inline test/e2e/common/container_probe.go]: These pods are all very similar. Reduce LOC by extracting to a factory method with the necessary parameters.

File: test/e2e/common/pods.go [hunk 2]
```diff
@@ -226,48 +209,9 @@ var _ = framework.KubeDescribe("Pods", func() {
 		})
 	})
 
-	It("should be schedule with cpu and memory limits [Conformance]", func() {
```
Comment (timstclair) [inline test/e2e/common/pods.go]: Why did you delete this test?

File: test/e2e/common/pods.go [hunk 3]
```diff
@@ -282,16 +226,6 @@ var _ = framework.KubeDescribe("Pods", func() {
...
-						Ports: []api.ContainerPort{{ContainerPort: 80}},
-						LivenessProbe: &api.Probe{
-							Handler: api.Handler{
-								HTTPGet: &api.HTTPGetAction{
-									Path: "/index.html",
-									Port: intstr.FromInt(8080),
-								},
-							},
-							InitialDelaySeconds: 30,
-						},
```
Comment (timstclair) [inline test/e2e/common/pods.go]: Why did you remove this?

File: test/e2e/common/pods.go (hunk 2 above)
Comment (Random-Liu) [inline test/e2e/common/pods.go]: Mentioned in the 1st comment of the PR:
It doesn't make sense at the node level.
It should have been tested in scheduler_predicates at the cluster level https://github.com/kubernetes/kubernetes/blob/master/test/e2e/scheduler_predicates.go#L264

:)

**Comment by vishh** [comment]
overall lgtm

**Comment by timstclair** [comment]
lgtm

File: test/e2e/common/pods.go (hunk 3 above)
Comment (Random-Liu) [inline test/e2e/common/pods.go]: Because this test doesn't need a liveness check, so remove dependency here to avoid the feature affecting irrelevant test. :)

File: test/e2e/common/container_probe.go (hunk 1 above)
Comment (Random-Liu) [inline test/e2e/common/container_probe.go]: ACK. Will send another PR to do that. :)
Thanks!
